dbt_meshify/storage/yaml_editors.py

```diff
@@ -259,36 +335,81 @@ def add_model_version(
         # create the new version file
 
         # if we're incrementing the version, write the new version file with a copy of the code
-        latest_version = (
-            int(self.model_node.latest_version) if self.model_node.latest_version else 0
-        )
-        last_version_file_name = (
-            f"{self.model_node.name}_v{latest_version}.{self.model_node.language}"
-        )
+        latest_version = int(self.node.latest_version) if self.node.latest_version else 0
+        last_version_file_name = f"{self.node.name}_v{latest_version}.{self.node.language}"
         next_version_file_name = (
-            f"{defined_in}.{self.model_node.language}"
+            f"{defined_in}.{self.node.language}"
             if defined_in
-            else f"{self.model_node.name}_v{latest_version + 1}.{self.model_node.language}"
+            else f"{self.node.name}_v{latest_version + 1}.{self.node.language}"
         )
-        model_path = self.get_model_path()
+        model_path = self.get_resource_path()
 
         if model_path is None:
-            raise Exception(f"Unable to find path to model {self.model_node.name}. Aborting.")
+            raise Exception(f"Unable to find path to model {self.node.name}. Aborting.")
 
         model_folder = model_path.parent
         next_version_path = model_folder / next_version_file_name
         last_version_path = model_folder / last_version_file_name
 
         # if this is the first version, rename the original file to the next version
-        if not self.model_node.latest_version:
+        if not self.node.latest_version:
             Path(self.project_path).joinpath(model_path).rename(
                 Path(self.project_path).joinpath(next_version_path)
             )
         else:
             # if existing versions, create the new one
-            self.file_manager.write_file(next_version_path, self.model_node.raw_code)
+            self.file_manager.write_file(next_version_path, self.node.raw_code)
             # if the existing version doesn't use the _v{version} naming convention, rename it to the previous version
-            if not model_path.root.endswith(f"_v{latest_version}.{self.model_node.language}"):
+            if not model_path.root.endswith(f"_v{latest_version}.{self.node.language}"):
                 Path(self.project_path).joinpath(model_path).rename(
                     Path(self.project_path).joinpath(last_version_path)
                 )
+
+    def move_resource(self):
+        """
+        move a resource file from one project to another
+
+        """
+        current_path = self.get_resource_path()
+        new_path = self.subdirectory / current_path
+        new_path.parent.mkdir(parents=True, exist_ok=True)
+        current_path.rename(new_path)
```

Comment:
My gut reaction is that this is a little odd. I suspect that it would be more ergonomic/useful to pass the destination path in as an argument. Having that been said, I'm only ~45% confident in this opinion.

Reply:
i don't disagree! i think it's mostly dependent on how much flexibility we want to afford users on where resources land in the new project